plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include "ContactPressureSensorsArrayDriver.h"
+#include <GazeboYarpPlugins/Handler.hh>
+#include <GazeboYarpPlugins/common.h>
+
+#include <yarp/os/LockGuard.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/math/Math.h>
+#include <yarp/math/SVD.h>
+
+#include <sdf/Param.hh>
+
+#define DEBUG 0
+#define N_AXIS 3
+#define TIME_CONVERSION_NS 1.0e-9
+
+using namespace yarp::dev;
+
+GazeboYarpContactPressureSensorsArrayDriver::GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+GazeboYarpContactPressureSensorsArrayDriver::~GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+void GazeboYarpContactPressureSensorsArrayDriver::onUpdate(const gazebo::common::UpdateInfo& _info)
+{ 
+    gazebo::msgs::Contacts contacts;
+    contacts = this->m_sensor->Contacts();
+    
+    ignition::math::Vector3d resultantForceonCG;
```

Comment:
Are you sure that this vectors are initialized to zero? From the context it seems that you are assuming this, but this is not documented. If this is actual the case ( https://bitbucket.org/ignitionrobotics/ign-math/src/5bf0933cc7325f6ccb149676a857bb725b163d35/include/ignition/math/Vector3.hh?at=default&fileviewer=file-view-default#Vector3.hh-58 ) it is a non documented behavior, so eventually they can change it. I would recommended to explicitly set the vector to zero to be future-safe.